kedro/io/core.py

```diff
@@ -601,8 +600,7 @@ def _get_save_path(self) -> PurePosixPath:
     def _get_versioned_path(self, version: str) -> PurePosixPath:
         return self._filepath / version / self._filepath.name
 
-    def load(self) -> _DO:
-        self.resolve_load_version()  # Make sure last load version is set
```

Comment:
This is actually quite interesting - pylint complains this is a useless parent delegation since the `load` method is exactly the same as `AbstractDataSet`. But let's address this in the refactoring PR instead. It feels like this is the initial design - thus the `load` method was necessary but later other changes come in, it become obsolete for a long time without anyone noticing.

Reply:
More context of this [Refactor PR of versioning](https://github.com/quantumblacklabs/private-kedro/pull/404) in 2+ years ago

Reply:
I am not sure about the point about extra protection - the function doesn't store any state in the dataset, so calling the function itself won't have any effect.

As the point of this function is hidden in a few levels of indirection - I think this should be more related to #1778

```python
    def resolve_load_version(self) -> Optional[str]:
        """Compute the version the dataset should be loaded with."""
        if not self._version:
            return None
        if self._version.load:
            return self._version.load
        return self._fetch_latest_load_version()
```